tick-api/src/main/java/me/aurium/tick/docker/machine/MachineSourceProvider.java

```diff
@@ -0,0 +1,91 @@
+package me.aurium.tick.docker.machine;
+
+import me.aurium.tick.docker.DockerSource;
+import me.aurium.tick.docker.DockerSourceProvider;
+import org.apache.commons.lang.SystemUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zeroturnaround.exec.ProcessExecutor;
+import org.zeroturnaround.exec.ProcessResult;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Pattern;
+
+public class MachineSourceProvider implements DockerSourceProvider {
+
+    private final String preferredDockerProvider;
+
+    private static final String DEFAULT = "default";
+
+    private final Logger logger = LoggerFactory.getLogger(MachineSourceProvider.class);
```

Comment:
Since you prefer non-static loggers, you can use `LoggerFactory.getLogger(getClass())` for a copy-and-paste proof line of code.